cosima_cookbook/explore.py

```diff
@@ -646,8 +646,13 @@ class DatabaseExplorer(VBox):
     ee = None
 
     def __init__(self, session=None, de=None):
+
+        if session is None:
+            session = database.create_session()
```

Comment:
Maybe this should be deferred to `DatabaseExtension` (and same for `ExperimentExplorer`)? Saves duplicating it thrice. Anywhere the Explorer needs to use `session` directly, e.g. when `ExperimentExplorer` gets the experiments, could be abstracted onto the Extension -- in that case specifically, you can just ask for `de.allexperiments`, right?

Reply:
Yeah, I think that is what I did initially, but for some reason forgot to save the `session` correctly at some point. I agree it shouldn't be done in all those places. 

Another possibility is to take it out of `DatabaseExtension`, as all this default stuff actually hid bugs. 

But I'll take your suggestion and do the default session opening only in `DatabaseExtension`.